### PR TITLE
fix: add pending_update tests + suppress jiti resolution noise

### LINKED ISSUE #395: fix: commit pending_update tests + suppress jiti resolution noise on exit
## Problem

Two minor issues discovered during PR #394 work:

### 1. Untracked test file
`tests/test_pending_update.py` was created during the pending_update GraphQL migration (PR #394) but never committed. Contains tests for `backfill_node_ids` and `run()`.

### 2. Noisy jiti path resolution on exit
When exiting pi, this debug message prints:
```
[async-agents] jiti path resolution failed: Cannot find module '@earendil-works/pi-coding-agent/package.json'
```

The `require.resolve` at `async-agents.ts:298` fails when pi is installed globally (npm global). The error is caught and handled gracefully (falls back correctly), but the `console.debug` message is noisy and confusing on every exit. Should be silenced or made conditional.

## Done

- [ ] Commit `tests/test_pending_update.py` to the test suite
- [ ] Suppress the jiti resolution debug message (remove console.debug or make it conditional on a verbose flag)

### PR DESCRIPTION
**Pull request by myakove**
## Summary

Two minor fixes:

### 1. Commit pending_update tests
`tests/test_pending_update.py` was created during PR #394 but never committed. Contains 13 tests covering `backfill_node_ids` (success, failure, edge cases, endpoint verification) and `run()` integration with backfill.

### 2. Suppress jiti path resolution noise
When exiting pi, `console.debug` printed a confusing error about `@earendil-works/pi-coding-agent/package.json` not being found. This is expected when pi is installed globally — the code falls back gracefully. Removed the noisy debug message.

Closes #395

### CODE REVIEW

File: extensions/orchestrator/async-agents.ts [hunk 1]
```diff
@@ -298,7 +298,7 @@ export function registerAsyncAgents(
       const piPkgDir = path.dirname(require.resolve("@earendil-works/pi-coding-agent/package.json"));
       const candidate = path.join(piPkgDir, "node_modules/jiti/lib/jiti-cli.mjs");
       if (fs.existsSync(candidate)) jitiCliPath = candidate;
-    } catch (e: any) { console.debug("[async-agents] jiti path resolution failed:", e?.message || e); }
+    } catch {}
```
[the remark this answers is not in the document]
Comment (myakove) [reply extensions/orchestrator/async-agents.ts]: Fixed: catch block now only suppresses MODULE_NOT_FOUND (expected for global install), logs all other errors via console.debug.

**Comment by myakove** [comment]
@qodo-code-review[bot]

The following review comments were reviewed and a decision was made:

### `extensions/orchestrator/async-agents.ts:298` (qodo bug) — Silent jiti resolution failure
> Addressed: Fixed: catch block now only suppresses MODULE_NOT_FOUND (expected for global install), logs all other errors via console.debug.

File: tests/test_pending_update.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        assert "node_id" not in comments[0]
+
+    def test_backfill_api_bad_json(self) -> None:
+        """Unparseable API response is handled gracefully."""
+        comments = [self._make_comment(comment_id=100, node_id=None)]
+        with patch("myk_pi_tools.reviews.pending_update.run_gh_api", return_value=None):
+            backfill_node_ids("owner", "repo", 1, 2, comments)
+
+        assert "node_id" not in comments[0]
+
```
[the remark this answers is not in the document]
Comment (myakove) [reply tests/test_pending_update.py]: Fixed: Consolidated 3 identical None-return tests into one, added separate test for non-list response covering the isinstance guard.

File: tests/test_pending_update.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+            }
+        ])
+        json_path = self._write_json(tmp_path, data)
+
+        # Mock update_comment_body to succeed
+        with patch("myk_pi_tools.reviews.pending_update.update_comment_body", return_value="success"):
+            run(json_path)
+
+        mock_backfill.assert_called_once_with("o", "r", 1, 2, data["comments"])
+
```
[the remark this answers is not in the document]
Comment (myakove) [reply tests/test_pending_update.py:192]: Fixed: Added call_order tracking with side_effect functions to assert backfill runs before update_comment_body.

**Comment by myakove** [comment]
@qodo-code-review[bot]

The following review comments were reviewed and a decision was made:

### `tests/test_pending_update.py:110` (qodo bug) — Misleading backfill error tests
> Addressed: Fixed: Consolidated 3 identical None-return tests into one, added separate test for non-list response covering the isinstance guard.
### `tests/test_pending_update.py:167` (qodo bug) — Backfill order not asserted
> Addressed: Fixed: Added call_order tracking with side_effect functions to assert backfill runs before update_comment_body.

File: extensions/orchestrator/async-agents.ts [hunk 4]
```diff
@@ -294,11 +294,29 @@ export function registerAsyncAgents(
...
+        if (piScript) {
+          const realPath = fs.realpathSync(piScript);
+          // pi script -> dist/cli.js, package root is 2 levels up
+          const piPkgDir = path.dirname(path.dirname(realPath));
+          const candidate = path.join(piPkgDir, "node_modules/jiti/lib/jiti-cli.mjs");
+          if (fs.existsSync(candidate)) jitiCliPath = candidate;
+        }
+      } catch (e: any) {
+        console.debug("[async-agents] jiti strategy 2 (process.argv) failed:", e?.message || e);
+      }
```
[the remark this answers is not in the document]
Comment (myakove) [reply extensions/orchestrator/async-agents.ts:318]: Fixed: Replaced console.debug with asyncLog() gated behind PI_ASYNC_DEBUG — no noise during normal runs, full logging when debug enabled.

**Comment by myakove** [comment]
@qodo-code-review[bot]

The following review comments were reviewed and a decision was made:

### `extensions/orchestrator/async-agents.ts:302` (qodo requirement gap) — console.debug logs jiti failures
> Addressed: Fixed: Replaced console.debug with asyncLog() gated behind PI_ASYNC_DEBUG — no noise during normal runs, full logging when debug enabled.

**Comment by myakove** [comment]
/ask The sticky comment still shows 'Misleading backfill error tests' as an unresolved bug, but we've already fixed the code — the three redundant tests (test_backfill_api_failure/timeout/bad_json) were consolidated into test_backfill_api_returns_none and a new test_backfill_api_returns_non_list was added to cover the isinstance guard. How do we get this finding cleared from the sticky summary?

**Comment by myakove** [comment]
## Findings Resolution (commit ed231d4)

**1. Misleading backfill error tests** — Fixed: Consolidated `test_backfill_api_failure/timeout/bad_json` into `test_backfill_api_returns_none` (single None-path test) + added `test_backfill_api_returns_non_list` (covers `isinstance` guard with dict input).

**2. Silent jiti resolution failure** — Fixed: Added dual-strategy resolution (require.resolve + process.argv[1] fallback). Errors logged via `asyncLog()` gated behind `PI_ASYNC_DEBUG`.

**3. Backfill order not asserted** — Fixed: `test_run_calls_backfill_before_update` uses `call_order` list with `side_effect` tracking, asserts `call_order.index("backfill") < call_order.index("update")`.

**4. console.debug logs jiti failures** — Fixed: Replaced `console.debug` with `asyncLog()` (PI_ASYNC_DEBUG-gated), no noise during normal runs.

All fixes landed in commits `005a0d9` through `ed231d4`. Please re-evaluate.

**Comment by myakove** [comment]
/ask What is the 'sticky-summary workflow' and how do I re-run it to clear stale findings from the sticky comment? Is there a specific command to refresh only the sticky summary without triggering a full review?
